src/BCFKeyValueData.cc

```diff
@@ -174,30 +174,58 @@ Status BCFKeyValueData::dataset_bcf_header(const string& dataset,
     return Status::OK();
 }
 
-// Note: if the bucket(s) does not exist, return NotFound.
-// If the bucket(s) exists, but no records were found, return an empty list
-Status BCFKeyValueData::dataset_bcf(const string& dataset,
-                                    const bcf_hdr_t* hdr,
-                                    const range& pos,
-                                    vector<shared_ptr<bcf1_t> >& records) const {
-    Status s;
+// Create a key for a bucket that includes records from [dataset],
+// on chromosome [rid], whose start position is in the genomic range [beg,end).
+//
+// Note that the end position might be outside the [beg,end) genomic range.
+// The search logic needs to compensate for this, by looking at several buckets.
+// The key is generated so that data for a genomic range from all samples will be
+// located contiguously on disk.
+static string gen_key(const string &dataset, const range& rng)
+{
+    return std::to_string(rng.rid) + ":" +
+        std::to_string(rng.beg) + ":" +
+        std::to_string(rng.end) + ":" +
+        dataset;
+}
```

Comment:
It would be nice to ensure the buckets are actually stored totally ordered by genomic range. This won't be the case here because the keys are sorted lexicographically, so for example 21:300000:3300000 precedes 21:90000:120000. This is not a dealbreaker though, feel free to file as a future todo.